docs/frontend-architecture.md

```diff
@@ -0,0 +1,27 @@
+# Frontend Architecture
+
+Major areas of the frontend code base include:
+
+- React Components (located in [src/components](../src/components) and [src/views](../src/views))
+  - These render the UI of the application. The views directory contains components that correspond to specific URLs of the app, and the components directory contains both UI components that are shared in many places, as well as the specific components that make up each section and subsection of the form (located in [/src/components/Section](../src/components/Section)). These Section components are where the actual questions and inputs for the form get rendered.
+  - All Section components get connected to Redux using the [SubsectionConnector](../src/components/Section/shared/SubsectionConnector.jsx) higher-order component. This passes in the relevant data for that section, any data needed from other sections, and top-level form logic (such as which questions are required based on the form type). It also passes in the `onUpdate` prop, which handles sending updated input values back up to Redux. Finally, an `onError` prop is also passed that saves validation Errors in Redux -- however, this is part of the legacy validation code, and should be deprecated as part of the Validation Refactor (UI) project (see the [frontend roadmap](frontend-roadmap.md)).
```

Comment:
```suggestion
  - All Section components get connected to Redux using the [SubsectionConnector](../src/components/Section/shared/SubsectionConnector.jsx) [higher-order component](https://reactjs.org/docs/higher-order-components.html). This passes in the relevant data for that section, any data needed from other sections, and top-level form logic (such as which questions are required based on the form type). It also passes in the `onUpdate` prop, which handles sending updated input values back up to Redux. Finally, an `onError` prop is also passed that saves validation Errors in Redux -- however, this is part of the legacy validation code, and should be deprecated as part of the Validation Refactor (UI) project (see the [frontend roadmap](frontend-roadmap.md)).
```

I've got a feeling "higher-order component might need some explaining so I added a link for the documentation.